src/OpenTelemetry.Extensions.Enrichment/TraceEnrichmentBag.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="TraceEnrichmentBag.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+public readonly struct TraceEnrichmentBag
+#pragma warning restore CA1815 // Override equals and operator equals on value types
+{
+    private readonly Activity activity;
+
+    public TraceEnrichmentBag(Activity activity)
+    {
+        this.activity = activity;
+    }
+
+    public void Add(string key, object? value)
+    {
+        this.activity.AddTag(key, value);
+    }
```

Comment:
I was thinking about this a bit. Generally we recommend people call `SetTag` not `AddTag` because `SetTag` is the OTel-compliant API. It will remove things when they are `null` and it won't create duplicates IIRC. Are you sure you want "Add" mechanics here? Also I think I brought this up earlier, but I don't see a whole lot of value in `TraceEnrichmentBag` might want to just expose the `Activity` directly and let users have control to do whatever they need 😄

Reply:
Activity was exposed initially, but as per [this ]( https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/949#discussion_r1114670788) discussion it was made private. Please reopen it there if you wish to continue?

Reply:
Also, replaced AddTag with SetTag